app/src/main/java/com/example/next_show/adapters/ShowAdapter.java

```diff
@@ -39,22 +44,45 @@ public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
 
     @Override
     public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
-        Show show = shows.get(position);
+        Show show = currentShows.get(position);
         holder.bind(show);
     }
 
     @Override
     public int getItemCount() {
-        return shows.size();
+        return currentShows.size();
     }
 
     public void clear() {
-        shows.clear();
+        currentShows.clear();
         notifyDataSetChanged();
     }
 
     public void addAll(List<Show> list) {
-        shows.addAll(list);
+        currentShows.addAll(list);
+        allShows = currentShows;
+        notifyDataSetChanged();
+    }
+
+    public void filter(String query){
+        // restore previous state of shows
+        currentShows = allShows;
+
+        // if query is blank, show all shows
+        if (!query.isEmpty()) {
+            Log.i("Adapter", "Filtering for " + query);
+            List<Show> queriedShows = new ArrayList<>();
+            for(Show s : currentShows){
+                if (s.getGenres().contains(query)) {
+                    queriedShows.add(s);
+                }
+            }
+
+            // update the adapter with shows that match the query
+            currentShows = queriedShows;
+        }
```

Comment:
Just documenting what we chatted about offline -- we can make this a bit dumber in the adapter to just pass it the new filtered list.  Which, it'll then just render.